mlflow/gateway/runner.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import os
+import subprocess
+import sys
+from typing import Generator
+from watchfiles import watch
+
+from mlflow.gateway import app
+from mlflow.gateway.config import _load_route_config
+from mlflow.gateway.utils import kill_child_processes
+
+
+_logger = logging.getLogger(__name__)
+
+
+def monitor_config(config_path: str) -> Generator[None, None, None]:
+    with open(config_path) as f:
+        prev_config = f.read()
+
+    for changes in watch(os.path.dirname(config_path)):
+        if not any((path == config_path) for _, path in changes):
+            continue
+
+        if not os.path.exists(config_path):
+            _logger.warning(f"{config_path} deleted")
+            continue
+
+        with open(config_path) as f:
+            config = f.read()
+        if config == prev_config:
+            continue
+
+        try:
+            _load_route_config(config_path)
+        except Exception as e:
+            _logger.warning("Invalid configuration: %s", e)
+            continue
+        else:
+            prev_config = config
+
+        yield
+
+
+class Runner:
+    def __init__(
+        self,
+        config_path: str,
+        host: str,
+        port: int,
+        workers: int,
+    ) -> None:
+        self.config_path = config_path
+        self.host = host
+        self.port = port
+        self.workers = workers
+        self.process = None
+
+    def start(self) -> None:
+        self.process = subprocess.Popen(
+            [
+                sys.executable,
+                "-m",
+                "gunicorn",
+                "--bind",
+                f"{self.host}:{self.port}",
+                "--workers",
+                str(self.workers),
+                "--worker-class",
+                "uvicorn.workers.UvicornWorker",
+                f"{app.__name__}:create_app_from_env()",
+                "--reload",
+                "--log-level",
+                "debug",
```

Comment:
This needs to be configurable.

Reply:
Agreed

Reply:
https://github.com/mlflow/mlflow/pull/8701